### PR TITLE
master -> main; use shiny-workflows

### PR DESCRIPTION
**Pull request by schloerke**
Related: https://github.com/rstudio/shiny/issues/3502

Also update to use reusable workflows

### CODE REVIEW

File: R/run-url.R [hunk 1]
```diff
@@ -23,10 +23,10 @@
 #' @examples
 #' ## Only run this example in interactive R sessions
 #' if (interactive()) {
-#'   runUrl('https://github.com/rstudio/shiny_example/archive/master.tar.gz')
```
Comment (schloerke) [inline R/run-url.R:26]: Switched here: https://github.com/rstudio/shiny_example

File: R/run-url.R [hunk 2]
```diff
@@ -121,7 +121,7 @@ runGist <- function(gist, destdir = NULL, ...) {
 #' @param username GitHub username. If `repo` is of the form
 #'   `"username/repo"`, `username` will be taken from `repo`.
 #' @param ref Desired git reference. Could be a commit, tag, or branch name.
-#'   Defaults to `"master"`.
+#'   Defaults to `"main"`.
```
Comment (schloerke) [inline R/run-url.R:124]: Is this ok?

File: tools/documentation/updateReexports.R [hunk 3]
```diff
@@ -16,14 +16,14 @@ local({
...
-  unlink(local_reexports_r_file)
-
   # pre document
   devtools::document()
 
   pre_namespace_lines <- readLines(rprojroot::find_package_root_file("NAMESPACE"))
 
+  local_reexports_r_file <- rprojroot::find_package_root_file("R/reexports.R")
+  unlink(local_reexports_r_file)
+
```
Comment (schloerke) [inline tools/documentation/updateReexports.R:26]: Must delete the reexports after collecting the `pre_namespace_lines`

File: tools/documentation/updateReexports.R [hunk 4]
```diff
@@ -119,13 +119,14 @@ local({
...
+  docs_are_same <- identical(pre_namespace_lines, post_namespace_lines)
+  msg_fn <- if (docs_are_same) message else stop
+  msg_fn(
     "\n",
-    "The NAMESPACE exports ", if (docs_have_changed) { "did NOT change"} else { "CHANGED"},
+    "The NAMESPACE exports ", if (docs_are_same) { "did NOT change"} else { "CHANGED"},
     " by copying in the ", pkg_names, " files\n",
     "\n",
-    if (docs_have_changed) "Possible ", pkg_names, " version requirement to add to DESCRIPTION file:\n",
+    if (docs_are_same) "Possible ", pkg_names, " version requirement to add to DESCRIPTION file:\n",
```
Comment (schloerke) [inline tools/documentation/updateReexports.R:129]: Now throws an error when the docs are not the same.

**Comment by schloerke** [comment]
Failing tests:

```
══ Failed tests ════════════════════════════════════════════════════════════════
── Failure (test-devmode.R:41:3): devmode can be turned on while _testing_ is disabled and check messages ──
`val <- get_devmode_option(key, ...)` did not throw the expected message.
Backtrace:
    █
 1. ├─shiny::with_devmode(...) test-devmode.R:41:2
 2. │ └─withr::with_options(...)
 3. │   └─base::force(code)
 4. └─shiny:::expect_devmode_option(...) test-devmode.R:55:4
 5.   └─testthat::expect_message(...) test-devmode.R:46:6
 6.     └─testthat:::expect_condition_matching(...)
── Failure (test-stacks.R:132:3): integration tests ────────────────────────────
df$num (`actual`) not equal to c(56L, 55L, 54L, 38L, 37L, 36L, 35L, 34L, 33L, 32L, 31L, 30L) (`expected`).

   `actual[7:9]`: 35 34 33         
`expected[7:12]`: 35 34 33 32 31 30
── Failure (test-stacks.R:134:3): integration tests ────────────────────────────
df$call (`actual`) not equal to c(...) (`expected`).

    actual                | expected                            
[7] "force"               | "force"                         [7] 
[8] "withVisible"         | "withVisible"                   [8] 
[9] "withCallingHandlers" | "withCallingHandlers"           [9] 
                          - "domain$wrapSync"               [10]
                          - "promises::with_promise_domain" [11]
                          - "captureStackTraces"            [12]
── Failure (test-stacks.R:137:3): integration tests ────────────────────────────
nzchar(df$loc) (`actual`) not equal to c(...) (`expected`).

`actual[7:9]`:    FALSE FALSE FALSE                  
`expected[7:12]`: FALSE FALSE FALSE FALSE FALSE FALSE
```

**Review by wch — APPROVED** [review]

File: R/run-url.R (hunk 2 above)
Comment (cpsievert) [reply R/run-url.R:124]: Maybe we could default this to `NA` and try to automatically detect if possible (maybe via usethis)? I feel like straight-up changing it to main is likely to break a lot of existing code

Comment (schloerke) [reply R/run-url.R]: Reverting change

File: .github/workflows/R-CMD-check.yaml [hunk 5]
```diff
@@ -1,139 +1,23 @@
-# NOTE: This workflow is overkill for most R packages
-# check-standard.yaml is likely a better choice
-# usethis::use_github_action("check-standard") will install it.
+# Workflow derived from https://github.com/rstudio/shiny-workflows
 #
-# For help debugging build failures open an issue on the RStudio community with the 'github-actions' tag.
-# https://community.rstudio.com/new-topic?category=Package%20development&tags=github-actions
+# NOTE: This Shiny team GHA workflow is overkill for most R packages.
+# For most R packages it is better to use https://github.com/r-lib/actions
```
Comment (cpsievert) [inline .github/workflows/R-CMD-check.yaml:4]:
```suggestion
# NOTE: The workflows below are designed specifically for RStudio's R packages. Other R packages should probably instead use `usethis::use_github_action_check_standard()` as a staring point
```

**Review by cpsievert — APPROVED** [review]

File: .github/workflows/R-CMD-check.yaml (hunk 5 above)
Comment (schloerke) [reply .github/workflows/R-CMD-check.yaml:4]: Too late. I'm not adjusting this for everyone

**Comment by schloerke** [comment]
Merging as it is no worse than main branch

**Pull request merged**